builder/dockerfile/dispatchers.go

```diff
@@ -71,6 +71,10 @@ func env(b *Builder, args []string, attributes map[string]bool, original string)
 			return errBlankCommandNames("ENV")
 		}
 
+		if runtime.GOOS == "windows" {
+			// Case insensitive environment variables on Windows
+			args[j] = strings.ToUpper(args[j])
```

Comment:
> It should only do the case insensitive comparison and leave the last key in its original form

@tonistiigi I disagree. As I've said environment variables are completely case insensitive on Windows. It's better to be consistent here and have docker manage the environment variables in single case stored in the config and where they're handled.

```
C:\> $env:FoObAr="something"
C:\> $env:foobar
something
C:\> $env:FOOBAR
something
```

But I do agree that commit.go also needs changing. Will update that.